### ISSUE TITLE
UserStore/RoleStore ConvertIdFromString can't use GUID

### ISSUE THREAD

**Issue by cherrydev** [aspnet/Identity#420]
``` (TKey)Convert.ChangeType(id, typeof(TKey)); ``` doesn't work for System.Guid

However, ``` (TKey) TypeDescriptor.GetConverter(typeof (TKey)).ConvertFromInvariantString(id); ``` does work!

**Comment by Bartmax** [comment]
is there any known workaround for this for the time being?

**Comment by cherrydev** [comment]
My workaround was implementing my own UserStore and RoleStore.  I was porting it to EF6 anyhow.

**Comment by suhasj** [comment]
@Bartmax you can override this method in your own subclass of UserStore/RoleStore and then hook that via DI in startup class. Let me know if you need more help with that

**Comment by Bartmax** [comment]
Thanks, this worked:
```
services.AddIdentity<ApplicationUser, ApplicationRole>()
    .AddUserStore<ApplicationUserStore>()
    .AddRoleStore<ApplicationRoleStore>();


public class ApplicationUserStore : UserStore<ApplicationUser, ApplicationRole, ApplicationDbContext, Guid>
    {
        public ApplicationUserStore(ApplicationDbContext context, IdentityErrorDescriber describer)
            : base(context, describer)
        {

        }
        public override Guid ConvertIdFromString(string id)
        {
            return new Guid(id);
        }
        public override string ConvertIdToString(Guid id)
        {
            return id.ToString();
        }
    }

    public class ApplicationRoleStore : RoleStore<ApplicationRole, ApplicationDbContext, Guid>
    {
        public ApplicationRoleStore(ApplicationDbContext context, IdentityErrorDescriber describer)
            :base(context, describer)
        {

        }
        public override string ConvertIdToString(Guid id)
        {
            return id.ToString();
        }
        public override Guid ConvertIdFromString(string id)
        {
            return new Guid(id);
        }
    }
```

**Comment by HaoK** [comment]
Was fixed in afff831d624b5a5e2677b0b42088626e8069336f
